CHANGELOG/v1.7.0.md

```diff
@@ -0,0 +1,433 @@
+## 👌 Kubernetes version support
+
+- Management Cluster: v1.26.x -> v1.29.x
+- Workload Cluster: v1.24.x -> v1.29.x
+
+[More information about version support can be found here](https://cluster-api.sigs.k8s.io/reference/versions.html)
+
+
+## Highlights
```

Comment:
Let's add the following instead:

* MachinePool: promoted to beta and enabled per default (#10141)
* MachineDeployment: Taint nodes with PreferNoSchedule during rollouts (#10223)
* MachineHealthCheck: implement annotation to manually mark machines for remediation (#10202)
* KCP improvements:
  * Enable kubeadm feature gates mutation (#10154)
  * Defer remediation when a control plane machine is still provisioning (#9734)
  * Prefer deletion of Machines with unhealthy control plane components during rollout (#10421)
* ClusterClass:
  * Add variable metadata (#10308)
  * Improved version validation if an update is already in progress (#10063)
  * Improved unit and e2e test coverage for ClusterClass with MachinePools (#10028)
* Various improvements to our e2e tests to increase test coverage and reduce flakes to improve our CI signal
* Implemented prowjob-gen to make it easier to manage our ProwJobs (#9937)